locust/test/test_main.py

```diff
@@ -29,6 +43,96 @@ def is_port_in_use(port: int) -> bool:
         return s.connect_ex(("localhost", port)) == 0
 
 
+def wait_for_output_condition_non_threading(
+    proc: Popen, output_lines: list[str], condition: str, timeout: int | float = 30
+) -> bool:
+    start_time = time.time()
+    while True:
+        combined_output = "\n".join(output_lines)
+        if condition in combined_output:
+            return True
+        if time.time() - start_time >= timeout:
+            return False
+        gevent.sleep(0.1)
+
+
+import gevent
+
+
+class PopenContextManager:
+    def __init__(self, args=None, file_content=None, port=None, **kwargs):
+        self.args = ["locust"] if args is None else args
+        self.kwargs = kwargs
+        self.process = None
+        self.output_lines = []
+        self.temp_file_path = None
+
+        if file_content is not None:
+            with NamedTemporaryFile(mode="w+", delete=False, suffix=".py") as temp_file:
+                temp_file.write(file_content)
+                temp_file.flush()
+                self.temp_file_path = temp_file.name
+            self.args += ["-f", self.temp_file_path]
+
+        if port is not None:
+            self.args += ["--web-port", str(port)]
+
+    def __enter__(self):
+        creation_flags = 0
+        # Add CREATE_NEW_PROCESS_GROUP for Windows
+        if platform.system() == "Windows":
+            creation_flags = subprocess.CREATE_NEW_PROCESS_GROUP
+
+        self.process = subprocess.Popen(
+            self.args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+            bufsize=1,
+            creationflags=creation_flags,  # Windows-specific creation flag
```

Comment:
these comments are a bit excessive. and you can do it as a oneliner to avoid having to store another variable:
`creationflags=subprocess.CREATE_NEW_PROCESS_GROUP if platform.system() == "Windows" else 0`